rbac/internal/views.py

```diff
@@ -77,9 +84,57 @@
 )
 from api.utils import RESOURCE_MODEL_MAPPING, get_resources
 
+# environment variables
+client_id = os.getenv("CLIENT_ID")
+client_secret = os.getenv("CLIENT_SECRET")
+
 logger = logging.getLogger(__name__)
 TENANTS = TenantCache()
 PROXY = PrincipalProxy()
+JWT = JWTCache()
+
+
+class SSOservice:
+    """Class to handle communication with SSO stage service."""
+
+    # Instance variable for the class.
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """Create a single instance of the class."""
+        if cls._instance is None:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+
+        return cls._instance
+
+    def __init__(self):
+        """Establish SSO connection information."""
+        self.conn = None
+
+    def get_conn(self):
+        """Get connection to sso stage."""
+        if settings.REDHAT_STAGE_SSO is not None:
+            self.conn = http.client.HTTPSConnection(settings.REDHAT_STAGE_SSO)
+        return self.conn
+
+
+sso_service = SSOservice()
+token = get_jwt_from_redis(
+    sso_service.get_conn(), settings.TOKEN_GRANT_TYPE, client_id, client_secret, settings.SCOPE, settings.OPENID_URL
+)
+
+
+@contextmanager
+def create_client_channel(addr):
+    """Create secure channel for grpc requests."""
+    # Call credential object will be invoked for every single RPC
```

Comment:
here could called method from your classes:

something like this (pseudo code):

jwt_cache = JWTCache()
...
jwt_token_provider = JWTTokenProvider()

jwt_token_manager = JwtTokenManager(jwt_cache, jwt_token_provider)
token = jwt_token_manager.get_token() # this will be new token or from redis